example/app/components/LoginScreen.tsx

```diff
@@ -3,53 +3,21 @@ import {View, Text, StyleSheet, TextInput, Pressable} from 'react-native';
 import colors from '../styles/colors';
 import {shadows} from '../styles/shadows';
 import {buttonStyles} from '../styles/button';
-import {Realm, useApp} from '@realm/react';
-
-export enum AuthState {
-  None,
-  Loading,
-  LoginError,
-  RegisterError,
-}
+import {AuthOperationName, useAuth, useEmailPasswordAuth} from '@realm/react';
 
 export const LoginScreen = () => {
-  const app = useApp();
+  const {result, logInWithEmailPassword} = useAuth();
+  const {register} = useEmailPasswordAuth();
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  const [authState, setAuthState] = useState(AuthState.None);
-
-  // If the user presses "login" from the auth screen, try to log them in
-  // with the supplied credentials
-  const handleLogin = useCallback(async () => {
-    setAuthState(AuthState.Loading);
-    const credentials = Realm.Credentials.emailPassword(email, password);
-    try {
-      await app.logIn(credentials);
-      setAuthState(AuthState.None);
-    } catch (e) {
-      console.log('Error logging in', e);
-      setAuthState(AuthState.LoginError);
-    }
-  }, [email, password, setAuthState, app]);
-
-  // If the user presses "register" from the auth screen, try to register a
-  // new account with the  supplied credentials and login as the newly created user
-  const handleRegister = useCallback(async () => {
-    setAuthState(AuthState.Loading);
-
-    try {
-      // Register the user...
-      await app.emailPasswordAuth.registerUser({email, password});
-      // ...then login with the newly created user
-      const credentials = Realm.Credentials.emailPassword(email, password);
-
-      await app.logIn(credentials);
-      setAuthState(AuthState.None);
-    } catch (e) {
-      console.log('Error registering', e);
-      setAuthState(AuthState.RegisterError);
-    }
-  }, [email, password, setAuthState, app]);
+
+  const handleLogin = useCallback(() => {
+    logInWithEmailPassword({email, password});
+  }, [logInWithEmailPassword, email, password]);
+
+  const handleRegister = useCallback(() => {
+    register({email, password});
+  }, [register, email, password]);
```

Comment:
I wonder if it makes sense to wrap these in `useCallback` or not ... I actually don't know how big a callback needs to be for it to be considered best practice to simply inline an arrow function instead 🤔

Reply:
It's not critical, but more of a best practice sort of thing.  Any function that's passed into a component should generally be wrapped in `useCallback` to ensure we aren't pushing an unneeded render into a child.  I think it's less about the performance of creating a function again, as it is holding onto a stable reference on re-renders

Reply:
I had discussed this with LJ.  If this were a web application with a form submit event, then we could setup a nice `useCallback` the was memoized and could get the form field values from the event, rather than the state.  Since this isn't possible in RN without writing our own formlike library (or using something else like Formik), using inline arrow functions is the path of least resistance.